pandas/core/groupby.py

```diff
@@ -1843,6 +1852,52 @@ def cummax(self, axis=0, **kwargs):
 
         return self._cython_transform('cummax', numeric_only=False)
 
+    def _get_cythonized_result(self, how, grouper, needs_mask=False,
+                               needs_ngroups=False, **kwargs):
+        """Get result for Cythonized functions
+
+        Parameters
+        ----------
+        how : str, Cythonized function name to be called
+        grouper : Grouper object containing pertinent group info
+        needs_mask : bool, default False
+            Whether boolean mask needs to be part of the Cython call signature
+        needs_ngroups : bool, default False
+            Whether number of groups part of the Cython call signature
+        **kwargs : dict
+            Extra arguments required for the given function. This method
+            internally stores an OrderedDict that maps those keywords to
+            positional arguments before calling the Cython layer
+
+        Returns
+        -------
+        GroupBy object populated with appropriate result(s)
+        """
+        exp_kwds = collections.OrderedDict([
+            (('group_fillna_indexer'), ('direction', 'limit')),
+            (('group_shift_indexer'), ('nperiods',))])
+
+        labels, _, ngroups = grouper.group_info
+        output = collections.OrderedDict()
+        base_func = getattr(libgroupby, how)
+
+        for name, obj in self._iterate_slices():
+            indexer = np.zeros_like(labels)
+            func = partial(base_func, indexer, labels)
+            if needs_mask:
+                mask = isnull(obj.values).astype(np.uint8, copy=False)
```

Comment:
Contrary to the int64 case, you do want to use `view` here - numpy doesn't see bool and uint8 as the 'same' type, so will always copy, which view elides.

Reply:
Thanks for the heads up - still trying to wrap my head around some of the nuances there. Out of curiosity, how do you know this? Just from experience or is there a good reference? 

I do see that `astype` documentation mentions `dtype, order and subok` requirements need to be satisfied to prevent copy, but I'm not clear on what those requirements are and if they are documented

Reply:
Unfortunately all I can really point to is trial and error.  Some "general" rules, at least in a pandas context
1) `copy=False` will only elide a copy if the dtype is identical (may be contradictions to this, but at least not in general).  Useful for conditional casts.
2) `arr.view(dtype)` never copies.  Viewing across types should only be used if if the itemsize of the two types are identical.  Primarily useful for casting a type with some metadata down to a a more primitive type (bool to uint8, datetime64 to int64).